cpu/lm4f120/periph/spi.c

```diff
@@ -0,0 +1,273 @@
+        }
+    },
+#endif
+};
+
+int spi_init_master(spi_t dev, spi_conf_t conf, spi_speed_t speed)
+{
+    unsigned long initialData = 0;
+    unsigned long mode = 0;
+    unsigned long bitrate;
+
+    if (dev >= SPI_NUMOF) {
+        return -1;
+    }
+
+    spi_poweron(dev);
+    ROM_SSIDisable(spi_confs[dev].ssi_base);
+    ROM_SysCtlPeripheralEnable(spi_confs[dev].gpio_sysctl);
+
+    ROM_GPIOPinConfigure(spi_confs[dev].pins.clk);
+    ROM_GPIOPinConfigure(spi_confs[dev].pins.fss);
+    ROM_GPIOPinConfigure(spi_confs[dev].pins.rx);
+    ROM_GPIOPinConfigure(spi_confs[dev].pins.tx);
+
+    ROM_GPIOPinTypeSSI(spi_confs[dev].gpio_port, spi_confs[dev].pins.mask);
+
+    switch (speed) {
+        case SPI_SPEED_100KHZ:
+            bitrate = 100000;
+            break;
+        case SPI_SPEED_1MHZ:
+            bitrate = 1000000;
+            break;
+        case SPI_SPEED_5MHZ:
+            bitrate = 5000000;
+            break;
+        case SPI_SPEED_10MHZ:
+            bitrate = 10000000;
+            break;
+        default:
+        case SPI_SPEED_400KHZ:
+            bitrate = 400000;
+            break;
+    }
+
+    /*
+       Polarity Phase        Mode
+       0     0   SSI_FRF_MOTO_MODE_0
+       0     1   SSI_FRF_MOTO_MODE_1
+       1     0   SSI_FRF_MOTO_MODE_2
+       1     1   SSI_FRF_MOTO_MODE_3
+     */
+    switch (conf) {
+        case SPI_CONF_FIRST_RISING: /**< first data bit is transacted on the first rising SCK edge */
+            mode = SSI_FRF_MOTO_MODE_0;
+            break;
+        case SPI_CONF_SECOND_RISING: /**< first data bit is transacted on the second rising SCK edge */
+            mode = SSI_FRF_MOTO_MODE_1;
+            break;
+        case SPI_CONF_FIRST_FALLING: /**< first data bit is transacted on the first falling SCK edge */
+            mode = SSI_FRF_MOTO_MODE_2;
+            break;
+        case SPI_CONF_SECOND_FALLING: /**< first data bit is transacted on the second falling SCK edge */
+            mode = SSI_FRF_MOTO_MODE_3;
+            break;
+    }
+
+    ROM_SSIClockSourceSet(spi_confs[dev].ssi_base, SSI_CLOCK_SYSTEM);
+    ROM_SSIConfigSetExpClk(spi_confs[dev].ssi_base, ROM_SysCtlClockGet(),
+                           mode,
+                           SSI_MODE_MASTER,
+                           bitrate,
+                           8);
+
+    ROM_SSIEnable(spi_confs[dev].ssi_base);
+
+    while (ROM_SSIDataGetNonBlocking(spi_confs[dev].ssi_base, &initialData)) ;
+    return 0;
+}
+
+int spi_init_slave(spi_t dev, spi_conf_t conf, char (*cb)(char))
+{
+    return -1;
+}
+
+void spi_transmission_begin(spi_t dev, char reset_val)
+{
+}
+
+int spi_acquire(spi_t dev)
+{
+    if (dev >= SPI_NUMOF) {
+        return -1;
+    }
+    mutex_lock(&locks[dev]);
+    return 0;
+}
+
+int spi_release(spi_t dev)
+{
+    if (dev >= SPI_NUMOF) {
+        return -1;
+    }
+    mutex_unlock(&locks[dev]);
+    return 0;
+}
+
+int spi_transfer_byte(spi_t dev, char out, char *in)
+{
+    unsigned long lin;
```

Comment:
ROM function writes a long even if we are only reading a char, so that's a `long in` :). I can add a comment, no problem.